src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -50,21 +51,42 @@ private Solution CreateSolutionWithProjectAndDocuments()
             var projectId = ProjectId.CreateNewId();
 
             return CreateSolution()
-                .AddProject(projectId, "goo", "goo.dll", LanguageNames.CSharp)
+                .AddProject(projectId, "proj1", "proj1.dll", LanguageNames.CSharp)
                 .AddDocument(DocumentId.CreateNewId(projectId), "goo.cs", "public class Goo { }")
                 .AddAdditionalDocument(DocumentId.CreateNewId(projectId), "add.txt", "text")
                 .AddAnalyzerConfigDocument(DocumentId.CreateNewId(projectId), "editorcfg", SourceText.From("config"), filePath: "/a/b");
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        private static IEnumerable<T> EmptyEnumerable<T>()
+        {
+            yield break;
+        }
+
+        // Returns an enumerable that can only be enumerated once.
+        private static IEnumerable<T> OnceEnumerable<T>(params T[] items)
+            => OnceEnumerableImpl(new StrongBox<int>(), items);
+
+        private static IEnumerable<T> OnceEnumerableImpl<T>(StrongBox<int> counter, T[] items)
+        {
+            Assert.True(counter.Value == 0);
+
+            foreach (var item in items)
+            {
+                yield return item;
+            }
+
+            counter.Value++;
```

Comment:
Note that if the enumeration is stopped in the middle, this never runs. Might be sanest to put the counter.Value++ before the foreach to avoid any odd bits slipping through?

Reply:
Sure.